### PR TITLE
Support subpixel render mode for mac font platform.

### PR DESCRIPTION
**Pull request by glennw**
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/537)

### CODE REVIEW

File: webrender/src/platform/macos/font.rs [hunk 1]
```diff
@@ -47,6 +49,31 @@ struct GlyphMetrics {
...
+    cg_context.set_rgb_fill_color(1.0, 1.0, 1.0, 1.0);
+    let point = CGPoint {x: -1., y: 0.};
+    let glyph = '|' as CGGlyph;
+    ct_font.draw_glyphs(&[glyph], &[point], cg_context.clone());
+    let data = cg_context.data();
+    let rgb : u32 = unsafe { mem::transmute::<[u8; 4], u32>([data[0],data[1],data[2],data[3]]) };
+    let r = (rgb >> 16) & 0xFF;
+    let g = (rgb >>  8) & 0xFF;
+    let b = (rgb >>  0) & 0xFF;
+    r != g || r != b
```
Comment (pcwalton) [inline webrender/src/platform/macos/font.rs]: Why is the u32 conversion needed if you just extract bytes again?

File: webrender/src/platform/macos/font.rs [hunk 2]
```diff
@@ -172,12 +211,18 @@ impl FontContext {
...
-                            ((rasterized_pixels[i * 4 + j] as f32) / alpha) as u8;
+
+                match render_mode {
+                    FontRenderMode::Alpha | FontRenderMode::Mono => {
+                        for i in 0..rasterized_area {
+                            let alpha = (rasterized_pixels[i * 4 + 3] as f32) / 255.0;
+                            for j in 0..3 {
+                                rasterized_pixels[i * 4 + j] =
+                                    ((rasterized_pixels[i * 4 + j] as f32) / alpha) as u8;
+                            }
```
Comment (pcwalton) [inline webrender/src/platform/macos/font.rs]: Could just be:

    for channel in &mut rasterized_pixels[(i*4)..(i*4 + 3)] {
        *channel = ((*channel as f32) / alpha) as u8
    }

**Comment by glennw** [comment]
Updated the cargo.lock file for wr-sample/replay.

**Pull request merged**